kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,240 @@
+      parametersFuture.addDoneListener {
+        val parameters: ExportTileCacheParameters = parametersFuture.get()
+        // export tiles to temporary cache on device
+        val exportTileCacheJob =
+          exportTileCacheTask.exportTileCache(parameters, exportTilesDirectory.path + "file.tpk")
+        exportTileCacheJob.start()
+        // create a progress dialog to show export tile progress
+        val dialog = createProgressDialog(exportTileCacheJob).apply {
+          getButton(DialogInterface.BUTTON_NEUTRAL).setTextColor(R.style.progressBarBlue)
+          // show the dialog
+          show()
+        }
+
+        // show progress of the export tile cache job on the progress bar
+        exportTileCacheJob.addProgressChangedListener {
+          dialog.progressBar.progress = exportTileCacheJob.progress
+        }
+
+        exportTileCacheJob.addJobDoneListener {
+          dialog.dismiss()
+          if (exportTileCacheJob.status == Job.Status.SUCCEEDED) {
+            val exportedTileCacheResult: TileCache = exportTileCacheJob.result
+            showMapPreview(exportedTileCacheResult)
+          } else {
+            Log.e(TAG, exportTileCacheJob.error.additionalMessage)
+            Toast.makeText(
+              this,
+              exportTileCacheJob.error.additionalMessage,
+              Toast.LENGTH_LONG
+            ).show()
+          }
+        }
+      }
+    }
+
+    // close the preview window
+    closeButton.setOnClickListener { clearPreview() }
+    clearPreview()
+  }
+
+  /**
+   * Show tile cache preview window including containing the exported tils.
+   *
+   * @param result takes the TileCache from the ExportTileCacheJob
+   */
+  private fun showMapPreview(result: TileCache) {
+    val newTiledLayer = ArcGISTiledLayer(result)
+    val previewMap = ArcGISMap(Basemap(newTiledLayer))
+
+    previewMapView.apply {
+      map = previewMap
+      setViewpoint(mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE))
+      visibility = View.VISIBLE
+      getChildAt(0).visibility = View.VISIBLE
+    }
+    mapPreviewLayout.bringToFront()
+    exportTilesButton.visibility = View.GONE
+  }
+
+  /**
+   * Create a progress dialog box for tracking the export tile cache job.
+   *
+   * @param exportTileCacheJob the export tile cache job progress to be tracked
+   * @return an AlertDialog set with the dialog layout view
+   */
+  private fun createProgressDialog(exportTileCacheJob: ExportTileCacheJob): AlertDialog {
+    val builder = AlertDialog.Builder(this@MainActivity)
+    with(builder) {
+      setTitle("Exporting tiles...")
+      // provide a cancel button on the dialog
+      setNeutralButton("Cancel") { _, _ ->
+        exportTileCacheJob.cancel()
+      }
+      setView(R.layout.dialog_layout)
+    }
+
+    return builder.create()
+  }
+
+  /**
+   * Clear the preview window.
+   */
+  private fun clearPreview() {
+    previewMapView.getChildAt(0).visibility = View.INVISIBLE
+    mapView.bringToFront()
```

Comment:
I made the original sample back when I didn't know alot about Android API -- is this call still needed?

Reply:
yup, `mapView.bringToFront` is still needed otherwise the preview pane gets in the way of the `mapView`: but I realised we can now do without the visibility call on `previewMapView`, thanks for raising this!